screens/Home/EventScreen/EventScreen.tsx

```diff
@@ -52,32 +77,54 @@ const EventScreen = () => {
     loadEvents()
   }, [])
 
-  const sections = [
-    { title: "Future Events", data: filteredFutureEvents },
-    { title: "Past Events", data: filteredPastEvents }
-  ]
-
-   const renderItem = ({ item }: SectionListRenderItemInfo<Event>) => (
-    //Need to add the onPress function to navigate to the event page
-    <TouchableOpacity>
-    <EventCard {...item} />
-    </TouchableOpacity>
-  ) 
-
-  const renderSectionHeader = (info : {section: typeof sections[number]} ) => (
-    <View style={{ backgroundColor: defautlBackgroundColor}} >
-    <Text style={styles.header}>{info.section.title}</Text>
-    <View style={styles.separationBar} />
+  function groupEventsByTwo(events: Event[]) {
+    const grouped = []
+    for (let i = 0; i < events.length; i += 2) {
+      // Check if there is a pair to push, if not push the last single event with a dummy event
+      if (i + 1 < events.length) {
+        grouped.push([events[i], events[i + 1]])
+      } else {
+        // Push the last item with a dummy transparent item
+        grouped.push([events[i], { uid: 'dummy', name: '', description: ''}])
+      }
+    }
+    return grouped
+  }
+  
+  
+
+
+  const renderItem = ({ item }: SectionListRenderItemInfo<Event[]>) => (
+    <View style={styles.row}>
+      {item.map((event) => (
+        <TouchableOpacity 
+          key={event.uid}  // Ensure each child has a unique key
+          style={[styles.cardContainer, event.uid === "dummy" ? styles.transparent : {}]}
+          disabled={event.uid === "dummy"}
+        >
+          <EventCard {...event} />
+        </TouchableOpacity>
+      ))}
     </View>
   )
+  
+
+  const renderSectionHeader = (info : {section: SectionListData<Event[], DefaultSectionT>} ) => {
```

Comment:
I feel like this way of writing is quite confusing... You could declare an interface like so: 

interface SectionHeaderProps {
  section: SectionListData<Event[], DefaultSectionT>;
} 

and when declaring the const do it like this: 
const RenderSectionHeader: React.FC <.SectionHeaderProps >  = ({ section }) => ....

Reply:
Ok